lib/internal/modules/cjs/loader.js

```diff
@@ -188,6 +189,11 @@ for (const { 0: id, 1: mod } of NativeModule.map) {
   }
 }
 
+let allBuiltins = builtinModules.slice();
+builtinModules.forEach((builtin) => allBuiltins.push(`node:${builtin}`));
+allBuiltins = new SafeSet(allBuiltins);
+NativeModule.getSchemeOnlyModuleNames().forEach((builtin) => allBuiltins.add(`node:${builtin}`));
```

Comment:
Nit: I would suggest keeping `allBuiltins` as a `Set`, rather than switching between `Array` and `Set`, `push` and `add`. This code snippet will avoid reassignment to `allBuiltIns`.

I have verified this suggestion locally.

```suggestion
const allBuiltins = new SafeSet(builtinModules);
builtinModules.forEach((builtin) => allBuiltins.add(`node:${builtin}`));
NativeModule.getSchemeOnlyModuleNames().forEach((builtin) => allBuiltins.add(`node:${builtin}`));
```

Reply:
```js
builtinModules.forEach((builtin) => allBuiltins.add(`node:${builtin}`));
```
^ will result in an [infinite loop](https://twitter.com/GNUmanth/status/1540027655847743488).

Reply:
It shouldn’t, because builtinModules and allBuiltins are different

Reply:
Ah, righto.

Reply:
I tried this suggestion on my local computer and made the test you created pass. I have not encounter the infinite loop problem.

Reply:
`allBuiltins.forEach` would have caused it.

Reply:
> It shouldn’t, because builtinModules and allBuiltins are different

@hemanth I think @ljharb has already answered why it would cause infinite loop.

``allBuiltins.forEach((builtin) => allBuiltins.add(`node:${builtin}`));``  will cause infinite loop, because it keep add element to itself while loop, so the loop will not end.

``builtinModules.forEach((builtin) => allBuiltins.add(`node:${builtin}`));``  will not cause infinite loop, because builtinModules and allBuiltins are different.

Reply:
Sorry, I read your word wrong. I thought you still don't believe it would not cause infinite loop. So I rephrase the reason. Ignore the above comment.😄

Reply:
```js
const allBuiltins = new SafeSet(safeFlatMap(builtinModules, (bm) => [bm, `node:${bm}`]));
NativeModule.getSchemeOnlyModuleNames().forEach((builtin) => allBuiltins.add(`node:${builtin}`));
```
How about something like^?

Reply:
It is great!